seacatauth/otp/service.py

```diff
@@ -119,46 +129,48 @@ async def unset_totp(self, credentials_id):
 
 
 	async def _create_totp_secret(self, session_id: str) -> str:
-		# Delete existing secret
+		"""
+		Create TOTP secret.
+		"""
+		# Delete secret if exists.
 		try:
-			await self._get_totp_secret(session_id)
 			await self._delete_totp_secret(session_id)
 		except KeyError:
 			# There is no secret associated with this user session
 			pass
 
-		upsertor = self.StorageService.upsertor(self.TOTPSecretCollection, obj_id=session_id)
+		upsertor = self.StorageService.upsertor(collection=self.TOTPUnregisteredSecretCollection, obj_id=session_id)
 
-		expires = datetime.datetime.now(datetime.timezone.utc) + self.RegistrationTimeout
+		expires: datetime.datetime = datetime.datetime.now(datetime.timezone.utc) + self.RegistrationTimeout
 		upsertor.set("exp", expires)
 
 		# TODO: Encryption
 		secret = pyotp.random_base32()
-		upsertor.set("__s", secret)
+		upsertor.set("__s", secret, encrypt=True)
 
-		await upsertor.execute()
+		await upsertor.execute(custom_data={"event_type": EventType.TOTP_CREATED})
```

Comment:
the enum string needs to be explicitly extracted: `EventType.TOTP_CREATED.name`. 

or if this feels too un-ergonomic, just use a plain class instead of the enum, as i did here for example:
https://github.com/TeskaLabs/seacat-auth/blob/main/seacatauth/openidconnect/utils.py#L1